packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts

```diff
@@ -363,9 +363,9 @@ export interface LambdaFunctionAssociation {
   readonly eventType: LambdaEdgeEventType;
 
   /**
-   * A version of the lambda to associate
+   * A version of the lambda to associate or a resolvable resource
    */
-  readonly lambdaFunction: lambda.IVersion;
+  readonly lambdaFunction: lambda.IVersion | cdk.IResolvable;
```

Comment:
We don't expose union types at the L2 layer.

Reply:
How should this be expressed then? Do you have an example?

Reply:
There is a pattern we follow, but I think you don't want that. See below.